conans/client/generators/cmake_common.py

```diff
@@ -323,8 +323,9 @@ def generate_targets_section(dependencies):
 endmacro()
 
 macro(conan_error_compiler_version)
-    message(FATAL_ERROR "Incorrect '${CONAN_COMPILER}' version 'compiler.version=${CONAN_COMPILER_VERSION}'"
-                        " is not the one detected by CMake: '${CMAKE_CXX_COMPILER_ID}=" ${VERSION_MAJOR}.${VERSION_MINOR}')
+    message(FATAL_ERROR "The installed dependencies are for compiler.version=${CONAN_COMPILER_VERSION},"
+        "but the compiler version CMake is using is ${VERSION_MAJOR}.${VERSION_MINOR}.\n"
+        "Please, check your default profile settings (conan profile show default) or your compiler (${CONAN_COMPILER})")
```

Comment:
This is good, but the error will also appear if not using the default profile but a custom profile, and the compiler doesn't match. Could you please put it in a generic way? Thanks!